include/sas.h

```diff
@@ -114,42 +110,46 @@ class SAS
   typedef uint64_t TrailId;
   typedef uint64_t Timestamp;
 
-#if HAVE_ZLIB_H
-  // Compression-related classes are only available if zlib is
+  // Compression profile
   class Profile
   {
   public:
-    inline Profile(std::string dictionary) : _dictionary(dictionary) {}
+    enum CompressionType {
+      ZLIB = 0,
+      LZ4
+    };
+
+    inline Profile(std::string dictionary, CompressionType t = ZLIB) : _dictionary(dictionary), _type(t) {}
+    inline Profile(CompressionType t) : _dictionary(""), _type(t) {}
```

Comment:
sas-client compiles with `-std=c++0x`, and (per the README) it needs to stay compatible with GCC 4.4 on RHEL, so I'll stick with the existing code.